src/OrleansRuntime/Placement/IPlacementDirector.cs

```diff
@@ -0,0 +1,25 @@
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.Placement
+{
+    /// <summary>
+    /// Interface for placement directors.
+    /// </summary>
+    internal interface IPlacementDirector
+    {
+        Task<PlacementResult> OnSelectActivation(
```

Comment:
I can give away I'd like to use in some places in the ADO.NET parts, also add `IAsyncEnumerable` and removing reflection parts (which Orleans operations do not use). :)